### PR TITLE
Migrated the tests (except integration tests which depend on Galvan, …

### PR DESCRIPTION
**Pull request by bbyreddy**
…Angela or Passthrough frameworks) to Junit 5.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -53,6 +53,12 @@
         <artifactId>guava</artifactId>
         <version>18.0</version>
       </dependency>
+      <dependency>
+        <groupId>org.junit.jupiter</groupId>
+        <artifactId>junit-jupiter-engine</artifactId>
+        <version>5.2.0</version>
+        <scope>test</scope>
```
Comment (lorban) [inline pom.xml]: There shouldn't be a scope in `dependencyManagement` dependencies.

File: connection-api/src/test/java/org/terracotta/exception/RuntimeEntityExceptionTest.java [hunk 2]
```diff
@@ -57,11 +57,11 @@ public void testEntityInfoProvided() {
...
-    Assert.assertNotEquals(description, impl.getMessage());
-    Assert.assertEquals(clz, impl.getClassName());
-    Assert.assertEquals(name, impl.getEntityName());
-    Assert.assertTrue(impl.getMessage().contains(clz));
-    Assert.assertTrue(impl.getMessage().contains(name));
+    Assertions.assertNotEquals(description, impl.getMessage());
+    Assertions.assertEquals(clz, impl.getClassName());
+    Assertions.assertEquals(name, impl.getEntityName());
+    Assertions.assertTrue(impl.getMessage().contains(clz));
+    Assertions.assertTrue(impl.getMessage().contains(name));
```
Comment (lorban) [inline connection-api/src/test/java/org/terracotta/exception/RuntimeEntityExceptionTest.java]: Static imports would look nicer.

File: pom.xml (hunk 1 above)
Comment (bbyreddy) [reply pom.xml]: Removed the scope from the dependencyManagement in this and other pom files as well.

File: connection-api/src/test/java/org/terracotta/exception/RuntimeEntityExceptionTest.java (hunk 2 above)
Comment (bbyreddy) [reply connection-api/src/test/java/org/terracotta/exception/RuntimeEntityExceptionTest.java]: Fixed

Comment (bbyreddy) [reply connection-api/src/test/java/org/terracotta/exception/RuntimeEntityExceptionTest.java]: Dependency on multiple versions of Junit will exists temporarily, only till we migrate the Galvan tests as well.

**Pull request closed without merging**